### DISCUSSION TITLE
Multichannel PWM for STM32U575

### DISCUSSION THREAD

**Discussion by bhavikbhansali** [zephyrproject-rtos/zephyr#53924] [Q&A]
I am trying to run a multichannel PWM for STM32U575 board. But I am not able to make it properly. I am new to zephyr so still few things I am learning and trying to know. Can any one help me with this?

**Overlayfile Details:**
![image](https://user-images.githubusercontent.com/11437390/213354778-044480cc-2f83-4441-9e2d-172db0023f3a.png)

Can any one help me modify the overlay file? Or suggest any of the example where multiple channel of same time is use to generate the PWM.

**Comment by FRASTM** [comment]
I checked with the disco_l475_iot1  and put an overlay as yours  using the timers15 and channel1. The output D10 of the CN1 (pin pa2) is toggling with variable duty cycle.

```
/ {
	pwmleds: pwmleds {
		compatible = "pwm-leds";
		status = "okay";

		pwmled4: pwmled4 {
			pwms = <&pwm15 1 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
		};
	};
};

&timers15 {
	status = "okay";
	st,prescaler = <1000>;
	pwm15: pwm {
		status = "okay";
		pinctrl-0 = <&tim15_ch1_pa2>; /* CN1 D10 */
		pinctrl-names = "default";
	};
};
```

**Comment by FRASTM** [comment]
Note that pwm15 channel 1 refers to &tim15_ch1, so that in _your_ DTS, if the pwm2 channel2 is enabled (on pa1),
 I expect `pwms = <&pwm2 2 PWM_MSEC(20) PWM_POLARITY_NORMAL>;`  in the pwmleds node.

**Reply by bhavikbhansali** [reply]
Thanks @FRASTM I am now able to make it working.
Below need to add into board specific overlay file.
`&timersX {`
`        st,prescaler = <1000>;`
`        status = "okay";`
`pwm: pwm {`
			`pinctrl-0 = <&timX_chY_P$& &timX_chY_P$&>;`
                        `pinctrl-names = "default";`
`		};`
`};`

X is timer number
Y is channel number
$ is GPIO port like A B C
& is pin number for specific port

this thing need to be define in overlay file of the project to make multiple PWM working.
